elisa/src/client/RunDetails/RunDetails.tsx

```diff
@@ -0,0 +1,245 @@
+        data: undefined,
+        dataError:  undefined,
+        columnInfo: {},
+        measures: [DEFAULT_X_AXIS_PROP, DEFAULT_Y_AXIS_PROP]
+    };
+
+    componentDidMount(): void {
+        const { schemaName, runId } = this.props.context;
+
+        Query.getQueryDetails({
+            schemaName,
+            queryName: 'Data',
+            success: (queryInfo) => {
+                const columnInfo = getColumnInfoFromQueryDetails(queryInfo);
+                this.setState(() => ({
+                    columnInfo: columnInfo.columnInfo,
+                    measures: columnInfo.measures
+                }), () => {
+                    this.queryData();
+                });
+            },
+            failure: (reason) => {
+                console.error(reason);
+                this.setState(() => ({ dataError: reason.exception }));
+            }
+        });
+
+        Query.selectRows({
+            schemaName,
+            queryName: 'Runs',
+            columns: RUN_COLUMN_NAMES.join(','),
+            filterArray: [Filter.create('RowId', runId)],
+            success: (data) => {
+                if (data?.rows?.length === 1) {
+                    this.setState(() => ({ runPropertiesRow: data.rows[0] }));
+                }
+                else {
+                    this.setState(() => ({ runPropertiesError: 'No properties found for run ID ' + runId + '.' }));
+                }
+            },
+            failure: (reason) => {
+                console.error(reason);
+                this.setState(() => ({ runPropertiesError: reason.exception }));
+            }
+        });
+    }
+
+    queryData() {
+        const { schemaName, runId } = this.props.context;
+        const { measures } = this.state;
+
+        Query.selectRows({
+            schemaName: schemaName,
+            queryName: 'Data',
+            columns: SAMPLE_COLUMN_NAMES.concat(measures).join(','),
+            filterArray: [Filter.create('Run/RowId', runId)],
+            success: (response) => {
+                this.setState(() => ({
+                    data: parsePlotDataFromResponse(response)
+                }));
+            },
+            failure: (reason) => {
+                console.error(reason);
+                this.setState(() => ({ dataError: reason.exception }));
+            }
+        });
+    }
+
+    render() {
+        const { runPropertiesError, dataError, runPropertiesRow, data } = this.state;
+
+        if (runPropertiesError || dataError) {
+            return <Alert>{runPropertiesError || dataError}</Alert>
+        }
+
+        if (!runPropertiesRow || !data) {
+            return <LoadingSpinner msg={'Loading data...'}/>
+        }
+
+        return <RunDetailsImpl {...this.props.context} {...this.state}/>
+    }
+}
+
+interface ImplProps {
+    protocolId: number,
+    runId: number,
+    runPropertiesRow: {[key: string]: any},
+    data: any[],
+    columnInfo: {[key: string]: any},
+    measures: string[]
+}
```

Comment:
You could carve out a common interface or type between these props and the state for the RunDetails.  Removes some duplicate code and makes sure they're in sync.

Reply:
I split out a CommonRunProps interface to extend in a few places.